### PR TITLE
Basic project architecture

### PR DESCRIPTION
**Pull request by fmontesino**
### This PR requires #1 to be merged first

Added a basic project architecture, featuring:

- Feature module separate from `app`.
- Hilt for Dependency Injection.
- ViewBinding enabled.
-  Minimal ViewModel implementation

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,13 +1,19 @@
...
     dependencies {
-        classpath "com.android.tools.build:gradle:4.2.0-alpha01"
-        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
+        classpath Libs.com_android_tools_build_gradle
+        classpath Libs.kotlin_gradle_plugin
+        classpath Libs.de_fayard_buildsrcversions_gradle_plugin
+        classpath Libs.detekt_gradle_plugin
+        classpath Libs.hilt_android_gradle_plugin
 
         // NOTE: Do not place your application dependencies here; they belong
```
Comment (kleinsenberg) [inline build.gradle]: I personally get rid of the extra noise from these comments, not sure if you like to do the same :)

File: feature-proxy-manager/proguard-rules.pro [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# Add project specific ProGuard rules here.
```
Comment (kleinsenberg) [inline feature-proxy-manager/proguard-rules.pro]: Maybe delete this and we'll add it if we actually need some rules?

File: settings.gradle [hunk 3]
```diff
@@ -1,2 +1,3 @@
 rootProject.name = "ProxyToggle"
 include ':app'
+include ':feature-proxy-manager'
```
Comment (kleinsenberg) [inline settings.gradle]: Second include not needed and we can comma separate?

**Review by kleinsenberg — APPROVED** [review]

File: build.gradle (hunk 1 above)
Comment (fmontesino) [reply build.gradle]: Can do yeah, as I don't visit this file often I generally don't care 😂

File: settings.gradle (hunk 3 above)
Comment (fmontesino) [reply settings.gradle]: 👍

File: feature-proxy-manager/proguard-rules.pro (hunk 2 above)
Comment (fmontesino) [reply feature-proxy-manager/proguard-rules.pro]: 👍

File: buildSrc/src/main/kotlin/Versions.kt [hunk 4]
```diff
@@ -30,8 +34,12 @@ object Versions {
 
     const val espresso_core: String = "3.2.0"
 
+    const val fragment_ktx: String = "1.2.5"
+
     const val lint_gradle: String = "27.2.0-alpha02"
 
+    const val viewbinding: String = "4.2.0-alpha02"
```
Comment (ChrisMTab) [inline buildSrc/src/main/kotlin/Versions.kt:41]: might as well use Compose too since you are using so many non final api libraries

**Pull request merged**